src/ui-test/tests/01_commands.projects.welcome.content.button.test.ts

```diff
@@ -0,0 +1,100 @@
+// /**
+//  * Licensed to the Apache Software Foundation (ASF) under one or more
+//  * contributor license agreements.  See the NOTICE file distributed with
+//  * this work for additional information regarding copyright ownership.
+//  * The ASF licenses this file to You under the Apache License, Version 2.0
+//  * (the "License", destination); you may not use this file except in compliance with
+//  * the License.  You may obtain a copy of the License at
+//  *
+//  *      https://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  * Unless required by applicable law or agreed to in writing, software
+//  * distributed under the License is distributed on an "AS IS" BASIS,
+//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  * See the License for the specific language governing permissions and
+//  * limitations under the License.
+//  */
+
+import * as fs from 'node:fs';
+import path from 'path';
+import {
+	ActivityBar,
+	CustomTreeSection,
+	EditorView,
+	InputBox,
+	ViewContent,
+	ViewControl,
+	VSBrowser,
+	WebDriver,
+	WelcomeContentSection
+} from 'vscode-extension-tester';
+import * as pjson from '../../../package.json';
+import {
+	killTerminal,
+	SPECIFIC_WORKSPACE_PATH,
+	TASK_FINISHED_IN_TERMINAL_TEXT,
+	waitUntilExtensionIsActivated,
+	waitUntilFileAvailable,
+	waitUntilTerminalHasText
+} from '../utils/testUtils';
+
+describe('Create a Camel Project using welcome content button', function () {
+	this.timeout(600000);
+	const WELCOME_CONTENT_BUTTON_WORKSPACE_PATH = path.join(SPECIFIC_WORKSPACE_PATH, 'using-welcome-content-button');
+	const WELCOME_CONTENT_BUTTON_WORKSPACE_NAME: string = 'using-welcome-content-button';
+	const CREATE_PROJECT_BUTTON = 'Create a Camel project';
+	const NO_FOLDER_OPENED_SECTION = 'No Folder Opened';
+	let driver: WebDriver;
+	let section: CustomTreeSection;
+	let content: ViewContent;
+	let input: InputBox;
+
+	before(async function () {
+		driver = VSBrowser.instance.driver;
+		fs.mkdirSync(WELCOME_CONTENT_BUTTON_WORKSPACE_PATH, { recursive: true });
+		await waitUntilExtensionIsActivated(driver, `${pjson.displayName}`);
+		const view = await ((await new ActivityBar().getViewControl('Explorer')) as ViewControl).openView();
+		content = view.getContent();
+		section = await content.getSection(NO_FOLDER_OPENED_SECTION);
+	});
+
+	after(async function () {
+		await ((await new ActivityBar().getViewControl('Explorer')) as ViewControl).closeView();
+		await new EditorView().closeAllEditors();
+		await killTerminal();
+	});
+
+	it(`Using button in section '${NO_FOLDER_OPENED_SECTION}'`, async function () {
+		const buttons = await ((await section.findWelcomeContent()) as WelcomeContentSection).getButtons();
+		// Find the correct button and click it
+		buttons.forEach(async (button) => {
+			if (await button.getTitle() === CREATE_PROJECT_BUTTON) {
+				await button.click();
+			}
+		});
```

Comment:
would be better to first search for the button and then click on it. it will avoid that it continue to resolve button titles or even worse to try to resolve title while the button is no more here because the button was clicked and the UI changed.

for another iteration: would be even better to have a getButton(title) on the API, reported https://github.com/redhat-developer/vscode-extension-tester/issues/1744

Reply:
I tried this approach before but it didnt work at the time. Now I know its because foreach doesnt work well with async methods :). I implemented a "find button by title" method in this test.